src/main/java/delta/util/Parser.java

```diff
@@ -0,0 +1,244 @@
+import delta.command.FindCommand;
+import delta.command.MarkCommand;
+import delta.command.PrintCommand;
+import delta.command.UnmarkCommand;
+import delta.exception.DeltaException;
+import delta.task.Deadline;
+import delta.task.Event;
+import delta.task.Todo;
+
+/**
+ * Deals with making sense of user input.
+ */
+public class Parser {
+    /**
+     * Formats a user typed date/time into proper format to be used by system.
+     *
+     * @param input User typed date/time.
+     * @return LocalDateTime object containing date/time obtained from input.
+     * @throws DeltaException If date/time input from user given in the wrong format.
+     */
+    private static LocalDateTime formatDateTime(String input) throws DeltaException {
+        try {
+            // Format to be provided by user input
+            DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm");
+            return LocalDateTime.parse(input, formatter);
+        } catch (DateTimeParseException e) {
+            throw new DeltaException("""
+                    OOPS!!! The format used for date/time is wrong!
+                    \t Please follow the proper format:
+                    \t yyyy-MM-dd HHmm
+                    \t eg. 2024-08-25 1800""");
+        }
+    }
+
+    /**
+     * Parses user input into a command understood by the system.
+     *
+     * @param input User typed command.
+     * @return Command to be used by system to execute an action.
+     * @throws DeltaException If format of user input is wrong.
+     */
+    public static Command parse(String input) throws DeltaException {
+        String[] description = input.strip().split(" ", 2);
+        String task = description[0];
+
+        // Bye
+        if (task.equalsIgnoreCase("bye")) {
+            return new ExitCommand();
+
+        // Print List
+        } else if (task.equalsIgnoreCase("list")) {
+            return new PrintCommand();
+
+        // Find Tasks
+        } else if (task.equalsIgnoreCase("find")) {
+            if (description.length == 2) {
+                String taskName = description[1].strip();
+                return new FindCommand(taskName);
+            } else {
+                throw new DeltaException("""
+                        OOPS!!! Description of task to find cannot be left blank!
+                        \t Please follow the proper format:
+                        \t * find [description]""");
+            }
+
+        // Add Todo
+        } else if (task.equalsIgnoreCase("todo")) {
+            if (description.length == 2) {
+                String taskName = description[1].strip();
+                return new AddCommand(new Todo(taskName));
+            // Only task given, no description
+            } else {
+                throw new DeltaException("""
+                        OOPS!!! Description of todo cannot be left blank!
+                        \t Please follow the proper format:
+                        \t * todo [description]""");
```

Comment:
Just a suggestion, but maybe it would be good to refactor some of this into the constructor of `DeltaException` because this message pattern is repeated a few times throughout `Parser.java`.